### PR TITLE
fix: snippet component

### PR DESCRIPTION
**Pull request by rflechtner**
## fixes a TODO

The snippets component implementation was highly brittle and not really useable. This regex-based implementation should be much more capable.

## How to test:

Check out the different examples where the snippet component or JsTsSnippet are used.

## Checklist:

- [x] I have verified that the code works
- [ ] I have verified that the code is easy to understand
  - [ ] If not, I have left a well-balanced amount of inline comments
- [ ] I have [left the code in a better state](https://deviq.com/principles/boy-scout-rule)
- [ ] I have documented the changes (where applicable)
    * Either PR or Ticket to update [the Docs](https://github.com/KILTprotocol/docs)
    * Link the PR/Ticket here

### CODE REVIEW

File: src/components/SnippetBlock/index.js [hunk 1]
```diff
@@ -4,60 +4,30 @@ import CodeBlock from '@theme/CodeBlock'
 const SnippetBlock = ({
   children,
   funcName = 'main',
-  funcEnd = '}',
-  snippets,
   leadingSpaces = 2,
+  dropHead = 0,
```
Comment (ntn-x2) [inline src/components/SnippetBlock/index.js:8]: Can you add a comment on what these two new parameters mean?

**Review by ntn-x2 — APPROVED** [review]
I guess you took the regex from somewhere online. In any case, I guess it does its job since the snippets look the same as the old ones. Cool improvement! I guess you can disregard my review in the other PR fixing the snippets since in this one everything is a `TsJsBlock` or snippet. There are few places where we use `CodeBlock` when we would probably have to use `TsJsBlock`, but that is out of scope for this PR. Maybe we can make a ticket for it?

**Comment by rflechtner** [comment]
> I guess you took the regex from somewhere online.

I'm saddened that you're doubting my regex proficiency! Actually built that myself :) 

It's either this or counting brackets, which may be even more robust.

> In any case, I guess it does its job since the snippets look the same as the old ones. 

I also ran a bunch of snippets against it in the regex playground. 

> There are few places where we use `CodeBlock` when we would probably have to use `TsJsBlock`, but that is out of scope for this PR. Maybe we can make a ticket for it?

Sure, let's do it

**Comment by ChrisChinchilla** [comment]
Thanks for the eyes @rflechtner !

File: src/components/SnippetBlock/index.js (hunk 1 above)
Comment (rflechtner) [reply src/components/SnippetBlock/index.js:8]: will do, they let you drop leading or trailing lines (e.g., if you want to not show the return statement).

**Review by ChrisChinchilla — APPROVED** [review]

File: docs/develop/07_dApp/04_verifier.md [hunk 2]
```diff
@@ -5,6 +5,7 @@ title: Verifying a Credential
 
 import TsJsSnippet from '@site/src/components/TsJsSnippet';
 import TsJsBlock from '@site/src/components/TsJsBlock';
+import TsJsSnippet from '@site/src/components/TsJsSnippet';
```
Comment (ChrisChinchilla) [inline docs/develop/07_dApp/04_verifier.md]:
```suggestion
```

**Pull request merged**